rpm/files_test.go

```diff
@@ -0,0 +1,76 @@
+package rpm
+
+import (
+	"context"
+	"testing"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/test"
+	"github.com/quay/zlog"
+)
+
+var rpmFilesTestcases = []struct {
+	name     string
+	isRPM    bool
+	filePath string
+}{
+	{
+		name:     "Found Node",
+		isRPM:    true,
+		filePath: "usr/lib/node_modules/npm/node_modules/safe-buffer/package.json",
+	},
+	{
+		name:     "Found Python",
+		isRPM:    true,
+		filePath: "usr/lib64/python3.9/site-packages/libcomps-0.1.18-py3.9.egg-info/PKG-INFO",
+	},
+	{
+		name:     "Not found",
+		isRPM:    false,
+		filePath: "usr/lib/node_modules/npm/node_modules/safe-buffer/package.jsonx",
+	},
+}
+
+func TestIsRPMFile(t *testing.T) {
+	ctx := zlog.Test(context.Background(), t)
+	a := test.NewCachedArena(t)
+	layer := test.LayerRef{
+		Registry: "registry.access.redhat.com",
+		Name:     "ubi9/nodejs-18",
+		Digest:   `sha256:1ae06b64755052cef4c32979aded82a18f664c66fa7b50a6d2924afac2849c6e`,
+	}
+	a.LoadLayerFromRegistry(ctx, t, layer)
+	r := a.Realizer(ctx).(*test.CachedRealizer)
+	t.Cleanup(func() {
```

Comment:
can we add another cleanup task (or something that will definitely run once all the subtests are done) which confirms the files cache is empty and the ref counts are all deleted (to test the gc)?

Reply:
So we can, but we can't guarantee that the `gc` go routine will complete before the test so we have two bad options:
* Add a waitgroup to the filesCache that only the test will call.
* [Sleep](https://cs.opensource.google/go/x/sync/+/refs/tags/v0.11.0:singleflight/singleflight_test.go;l=125)

Reply:
I went down the first route